add migration guide and cheat sheet (#873)
iotaledger/documentation: getting-started/1.5/chrysalis/migration_guide.md

# Developer Guide to Chrysalis (IOTA 1.5)

## What is new?
In the Second Part of Chrysalis, alot of things changed. Let's discover, what in detail:

- Ed25519 support— a more commonly used signature scheme and support for reusable addresses.
- UTXO— A better fund and state management approach.
- Binary transaction layout— validation, IO and other processing working on binary data without the need to have many binary-ternary conversions as in the IOTA 1.0 node software.
- Atomic transactions— Move from the current Bundle construct and use of simpler atomic transactions instead
- Autopeering
- Whiteflag
- Improved Tipselection
- New Milestone Selection
- Dust Protection
- Up to 8 Parents

## Node Software

- [hornet (branch: develop)](https://github.com/gohornet/hornet/tree/develop)
- [bee (branch: chrysalis-pt-2)](https://github.com/iotaledger/bee/tree/chrysalis-pt-2)

## Libraries

> - [iota.rs](https://github.com/iotaledger/iota.rs)

Reviewer: https://github.com/iotaledger/iota.rs/tree/dev/bindings/python is there as well now

Reviewer 2: Should iota.c and iota.js also be listed?

Author: @fijter Add also iota.c and iota.js?

i would like to add iota.c - and also iota.js with a bit explanation. 

Because if we just add it, develpers could be defused about 2 libs for javascript. We need to explain the different between neon bindings and iota.js in a simple way and which one we recommend. 

What do you think?

Reviewer: iota.c would be good, I'm doubting about iota.js because it is indeed confusing and it's not the library we want to promote over the rust bindings one. I think people will have a hard enough time finding the nodejs bindings version in the iota.rs repo. Maybe we can add it but either in small text or with a big notification above it to use iota.rs unless you can't for some reason.

    - [Node.js](https://github.com/iotaledger/iota.rs/tree/dev/bindings/node)
- [wallet.rs](https://github.com/iotaledger/wallet.rs)
    - [Node.js](https://github.com/iotaledger/wallet.rs/tree/develop/bindings/nodejs)

## Infrastructure
Nodes deployed to the testnet can be queried using a load balancer at:

- api.lb-0.testnet.chrysalis2.com

We recommend using the load balancer for most scenarios.

> Single node endpoints, for example for MQTT use, are:

Reviewer: ```suggestion
Single node endpoints that expose native MQTT in case you need this are:
```

- api.hornet-0.testnet.chrysalis2.com
- api.hornet-1.testnet.chrysalis2.com
- api.hornet-2.testnet.chrysalis2.com
- api.hornet-3.testnet.chrysalis2.com

Node API is integrated according to the [following specification](https://editor.swagger.io/?url=https://raw.githubusercontent.com/rufsam/protocol-rfcs/master/text/0026-rest-api/rest-api.yaml)

## Developer Tools
- [Explorer](https://explorer.iota.org/chrysalis)
- [Online Faucet](https://faucet.testnet.chrysalis2.com/)
- [cli-wallet](https://github.com/iotaledger/cli-wallet)
- [chrysalis-faucet Code (nodejs + svelte)](https://github.com/iotaledger/chrysalis-faucet)

## A note about seeds and addresses

> In Crysalis, all Ternary things has been removed. which results in a better developer expirence.

Reviewer: ```suggestion
In Chrysalis all ternary conversions have been removed, which results in a better and faster developer experience.
```

Also the WOTS-Signarure has beed chaged to the Ed25519 Signarure scheme. That means, you can use an address now multiple times to send and recive tokens. 

So all these changes results into a complete different view of addresses and seeds: 

IOTA 1.0 Address

```bash=
UDYXTZBE9GZGPM9SSQV9LTZNDLJIZMPUVVXYXFYVBLIEUHLSEWFTKZZLXYRHHWVQV9MNNX9KZC9D9UZWZRGJMIGPDW
```

IOTA Chrysalis Address

```bash=

> fe1b388f81c70b7603fda53e14af7b6dfd916f39a51ea83352cc771385e1d064

Reviewer 2: Also add the bech32 format

Reviewer: Yes please, bech32 should be the public default, the hex address format is just for internal reference and low-level usage.

```

With the new wallet library, developers dont need to usa a Seed anymore. The Seed is backed up with Stronghold, which is an software security box. It's not possible to get the seed out of this box. It uses encrypted snapshots that can be easily backed up and securely shared between devices. These snapshots are secured by an password.

Read more about Stronghold on the [Stronghold docs page](https://stronghold.docs.iota.org).

## A note about Coordicide (IOTA 2.0)

- Do we expect breaking changes to client libs like iota.rs or wallet.rs for the Coordicde Update?

> todo

## Deep Dive

### Dust Protection

> What is Dust? We define the spamming of low value transaction as dust. 

Reviewer 2: Maybe to be more clear write <1 Mi

> In the UTXO model, each node in the network needs to keep track of all the currently unspent outputs. When the number of outputs gets too large, this can cause performance and memory issues. This RFC proposes a new protocol rule regarding the processing of outputs that transfer a very small amount of IOTA, so-called dust outputs: Dust outputs are only allowed when they are backed up by a certain deposit on the receiving address. This limits the amount of dust outputs, thus making it expensive to proliferate dust. Since a receiver must make a deposit, the protocol makes receiving dust an opt-in feature.

[protocol-rfcs#0032](https://github.com/iotaledger/protocol-rfcs/pull/32)

### Up to 8 Parents.

> An message can be bind to minimal 2 parents and up to 8 parents. This rusults in a ...

Reviewer 2: A message can reference 1-8 parent messages

:::danger
**Question**

> Which advantates has the "8 Parent". RFC?

Reviewer: ```suggestion
Which advantages has the "8 Parent". RFC?
```
